docs/meshstack.azure.index.md

```diff
@@ -395,3 +396,112 @@ The arguments available here are:
 3. argument: meshProject [ID (numeric)](./meshstack.configuration.md#identifiers)
 4. argument: tenantNumber (numeric), a running number specific to each platform which can optionally be enabled
 5. argument: role name suffix ([configurable via Landing Zone](./meshstack.azure.landing-zones.md#meshrole-to-platform-role-mapping))
+
+## Permission Model
+<!-- markdownlint-disable MD001 -->
+
+meshstack in total needs up to three service principals, one for replicating meshProjects into the Azure platform, one if AAD should be used as an user lookup when users are added to a customer or project inside the meshfed panel and one for [gathering metering data](./meshstack.azure.metering.md) for the billing subsystem.
+
+#### meshReplicator Project Provisioning
+
+- `Directory.Read.All` - this permission is required to search the directory for existing users, groups and service principals
+- `Group.ReadWrite.All` - this permissions is required to create new groups
+- `User.Invite.All` - this permission is required if you want to enable B2B User Invitations
+- `User Access Administrator` - to add and remove user group access to the Subscriptions
+- (optional)`Permissions to call Azure Functions` - see the [end of this section](#azure-function-access-setup) for how to setup this
+
+Created Subscriptions will have the Service Principal of the replicator registered as an owner at first. As soon as all needed maintenance steps are performed (e.g. renaming the subscription, moving it into the final management group), the replicator removes itself as an owner.
+
+All permissions left are therefore granted only via the management group hierarchy. The meshstack software does **not** need access related to actual workload inside these subscriptions. However in order to perform certain maintenance tasks, the following permissions/roles must be granted to the replicator principal:
+
+- `Blueprint Operator` - required for assigning and managing Blueprint assignments to the Subscriptions
+- `Management Group Contributor` - required on the target management group for moving Subscriptions
+
+In order to prevent that the replicator from assigning itself more permissions than originally anticipated, it is recommended to add the following policy rule:
+
+```json
+{
+    "properties": {
+        "mode": "all",
+        "displayName": "meshcloud Privilege Escalation Prevention",
+        "description": "Prevent meshcloud SP from assigning itself new roles.",
+        "policyRule": {
+          "if": {
+            "allOf": [
+              {
+                "equals": "Microsoft.Authorization/roleAssignments",
+                "field": "type"
+              },
+              {
+                "allOf": [
+                  {
+                    "field": "Microsoft.Authorization/roleAssignments/roleDefinitionId",
+                    "equals": "/subscriptions/<SUBSCRIPTION_ID>/providers/Microsoft.Authorization/roleDefinitions/<AZURE_SERVICE_PRINCIPAL_ROLE_ID>"
```

Comment:
d: This may be wrong, i.e. AZURE_SERVICE_PRINCIPAL_ROLE_ID is the **expected** role assignment. Shouldn't this policy audit any **unexpected** role assignments? should this be `notEquals` instead of `equals` (check with azure policy language, I'm not 100% familiar with it)